app/uk/gov/hmrc/economiccrimelevyregistration/controllers/RegistrationSubmissionController.scala

```diff
@@ -16,129 +16,118 @@
+                    )
+                    .asResponseError
+      _         = if (appConfig.amendRegistrationNrsEnabled) {
+                    nrsService.submitToNrs(
+                      registration.base64EncodedFields.flatMap(_.nrsSubmissionHtml),
+                      response.eclReference,
+                      appConfig.eclAmendRegistrationNotableEvent
+                    )
                   }
+      _         = auditService.successfulSubscriptionAndEnrolment(registration, response.eclReference, liabilityYear)
+    } yield response
 
-                case Valid(Right(registration)) =>
-                  dmsService
-                    .submitToDms(registration.base64EncodedFields.flatMap(_.dmsSubmissionHtml), Instant.now())
-                    .flatMap {
-                      case Right(response) =>
-                        auditService
-                          .successfulSubscriptionAndEnrolment(
-                            registration,
-                            response.eclReference,
-                            additionalInfo.liabilityYear
-                          )
-                        Future.successful(Ok(Json.toJson(response)))
-                      case Left(e)         =>
-                        logger.error(
-                          s"Failed to submit PDF to DMS: ${e.getMessage()}"
-                        )
-                        Future.successful(InternalServerError("Could not send PDF to DMS queue"))
-                    }
-                case Invalid(e)                 =>
-                  logger.error(
-                    s"Invalid registration: ${e.toList.mkString(",")}"
-                  )
-                  Future.successful(InternalServerError(Json.toJson(DataValidationErrors(e.toList))))
-              }
-          )
-      case None               => Future.successful(NotFound)
+  def subscribeToEcl(registration: Registration, additionalInfo: RegistrationAdditionalInfo)(implicit
+    hc: HeaderCarrier,
+    request: AuthorisedRequest[_]
+  ): EitherT[Future, ResponseError, CreateEclSubscriptionResponsePayload] =
+    for {
+      sub      <- subscriptionValidation(registration, additionalInfo).asResponseError
+      response <- subscriptionService.subscribeToEcl(sub, registration, None).asResponseError
+      _         = if (appConfig.nrsSubmissionEnabled) {
+                    nrsService
+                      .submitToNrs(
+                        registration.base64EncodedFields.flatMap(_.nrsSubmissionHtml),
+                        response.success.eclReference,
+                        appConfig.eclAmendRegistrationNotableEvent
+                      )
+                      .asResponseError
+                  }
+    } yield response.success
+  private def subscriptionValidation(
+    registration: Registration,
+    additionalInfo: RegistrationAdditionalInfo
+  ): EitherT[Future, DataValidationError, EclSubscription]                =
+    EitherT {
```

Comment:
can avoid this by returning EitherT in validateSubscription

Reply:
Fixed :)